packages/amplication-client/src/Purchase/PurchasePage.tsx

```diff
@@ -76,17 +81,34 @@ const PurchasePage = (props) => {
     history.action !== "POP" ? history.goBack() : history.push("/");
   }, [history]);
 
-  const [provisionSubscription, { loading: provisionSubscriptionLoading }] =
-    useMutation<DType>(PROVISION_SUBSCRIPTION, {
-      onCompleted: (data) => {
-        const { provisionStatus, checkoutUrl } = data.provisionSubscription;
-        if (provisionStatus === "PAYMENT_REQUIRED")
-          window.location.href = checkoutUrl;
-      },
-      onError: (error: ApolloError) => {
-        console.log(error);
-      },
-    });
+  const [
+    provisionSubscription,
+    {
+      loading: provisionSubscriptionLoading,
+      error: provisionSubscriptionError,
+    },
+  ] = useMutation<DType>(PROVISION_SUBSCRIPTION, {
+    onCompleted: (data) => {
+      const { provisionStatus, checkoutUrl } = data.provisionSubscription;
+      if (provisionStatus === "PAYMENT_REQUIRED") {
+        window.location.href = checkoutUrl;
+      } else if (provisionStatus === "SUCCESS") {
```

Comment:
Ideally we should use the Enum for provisionStatus and not an hardcoded string.

Update server to return the correct enum in `ProvisionSubscriptionResult` as follow:

```ts
import { Field, ObjectType, registerEnumType } from "@nestjs/graphql";
import { ProvisionSubscriptionStatus } from "@stigg/node-server-sdk";

@ObjectType({
  isAbstract: true,
})
export class ProvisionSubscriptionResult {
  @Field(() => ProvisionSubscriptionStatus, {
    nullable: false,
  })
  provisionStatus!: ProvisionSubscriptionStatus;

  @Field(() => String, {
    nullable: true,
  })
  checkoutUrl?: string;
}

registerEnumType(ProvisionSubscriptionStatus, {
  name: "ProvisionSubscriptionStatus",
});
```

Reply:
I agree with the need but prefer not to go with this change in the current scope. 
I will keep the existing code as it is, and we can consider making the change in a separate scope after considering the impact and different options

Reply:
I opened a new issue
https://github.com/amplication/amplication/issues/8639